test/spmd/tensor/test_redistribute.py

```diff
@@ -165,6 +165,27 @@ def test_replicate_to_partial(self):
                 partial_tensor.to_local(), torch.zeros_like(local_tensor)
             )
 
+        # replicate to partial on sub groups
+        subgroup = new_group(ranks=[1, 3])
+        device_mesh = DeviceMesh(self.device_type, [1, 3], dim_groups=subgroup)
+        # 1) test replicate -> partial on subgroup
+        replica_tensor = DTensor(
+            local_tensor, device_mesh, replica_spec, requires_grad=True
+        )
+        partial_tensor = Redistribute.apply(
+            replica_tensor, device_mesh, partial_spec
+        )
+        self.assertEqual(partial_tensor.size(), local_tensor.size())
+
+        if self.rank == 1:
```

Comment:
What about when self.rank == 3?

Reply:
for replicate to partial, we zero out all other ranks content, so in this case `self.rank == 3` should be zero tensor. But I guess for rank=1 and rank=2 we shouldn't zero out, let me see what's going on